### ISSUE TITLE
Disable SSL Certificate Verification

### ISSUE THREAD

**Issue by spacemishka** [sherlock-project/sherlock#1388]
- [X] I'm reporting a feature request
- [X] I've checked for similar feature requests including closed ones

Could you implement a switch to disable the verification of SSL Certificates? It could be useful behind some proxies. Currently i am facing this issue

` (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:997)')))
Using the proxy: `

thanks!

**Comment by Sebhadde** [comment]
Hh

**Comment by spacemishka** [comment]
> Hh

sorry?

**Comment by DavidEl03** [comment]
Hi spacemishka

What website gave you the error?
Are you able to recreate this error reliably?

**Comment by spacemishka** [comment]
I get this with any website and i is always happen when running behind a proxy. NO issues when using Python requests with ssl verification disabled.

**Comment by DavidEl03** [comment]
I think that the best course of action is to add an argument that disables SSL checks in requests
that argument should be accessible from where [the request is being sent](https://github.com/sherlock-project/sherlock/blob/a4c0fb05aa1873d06c3740b7f19df82c90a814d4/sherlock/sherlock.py#L310)

and pass the boolean into an argument called `verify`. when `verify = False` requests wont check SSL integrity

I will work on that if I get a chance, but feel free to take over this.

**Comment by 0xStryK3R** [comment]
Hi @DavidEl03 , I would like to take this up if this issue is still pending. Do assign it to me if it is open for contribution.

**Comment by DavidEl03** [comment]
@0xStryK3R feel free to implement the changes, I think it's still up for grabs. Just double check if it's still an issue by using the tool with VPN or proxy

**Comment by Dragonblue96** [comment]
> * [x]  I'm reporting a feature request
> 
>     * [x]  I've checked for similar feature requests including closed ones
> 
> 
> Could you implement a switch to disable the verification of SSL Certificates? It could be useful behind some proxies. Currently i am facing this issue
> 
> `(Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:997)'))) Using the proxy:`
> 
> thanks!

how did you fix this issue?
Please help me I'm facing the similar issue

**Comment by sdushantha** [comment]
This is an issue with your environment and not Sherlock itself. I've encountered similar issues with cURL before and was able to fix it systemwide. Unsure what I did but did find it on Stackoverflow somewhere

**Comment by 0xdavidel** [comment]
> This is an issue with your environment and not Sherlock itself. I've encountered similar issues with cURL before and was able to fix it systemwide. Unsure what I did but did find it on Stackoverflow somewhere

Hey bud, this is definitely a needed feature and a real issue, changing system wide configuration just for a single tool is not a viable solution. It's just an opening for trouble down the road for those users.

**Comment by sdushantha** [comment]
Yeah thats a good point, didn't think about that. I've opened this issue and if @0xStryK3R still wants to work on this, I'll take a look at the PR :+1:

**Comment by ll299** [comment]
Hi, our team is working on this.
